sql/descriptor.go

```diff
@@ -111,12 +113,22 @@ func (p *planner) createDescriptor(plainKey descriptorKey, descriptor descriptor
 	b.CPut(descKey, descriptor, nil)
 	// Mark transaction as operating on the system DB.
 	p.txn.SetSystemDBTrigger()
-	return p.txn.Run(&b)
+	if err := p.txn.Run(&b); err != nil {
+		return err
+	}
+	dcache[string(key)] = descriptor
```

Comment:
Peter's change is not going to be merged. I'm working on using the gossiped system DB as a descriptor cache.